src/back/utils/contractInteractions.ts

```diff
@@ -0,0 +1,157 @@
+import { abi as BadgesAbi } from '@otterspace-xyz/contracts/out/Badges.sol/Badges.json'
+import logger from 'decentraland-gatsby/dist/entities/Development/logger'
+import { ethers } from 'ethers'
+
+import { POLYGON_BADGES_CONTRACT_ADDRESS, RAFT_OWNER_PK, TRIMMED_OTTERSPACE_RAFT_ID } from '../../constants'
+import { ActionStatus, BadgeCreationResult, GAS_MULTIPLIER, GasConfig } from '../../entities/Badges/types'
+import RpcService from '../../services/RpcService'
+import { AirdropJobStatus, AirdropOutcome } from '../types/AirdropJob'
+
+const TRANSACTION_UNDERPRICED_ERROR_CODE = -32000
+
+function checksumAddresses(addresses: string[]): string[] {
+  return addresses.map((address) => ethers.utils.getAddress(address))
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function estimateGas(estimateFunction: (...args: any[]) => Promise<any>): Promise<GasConfig> {
+  const provider = RpcService.getPolygonProvider()
+  const gasLimit = await estimateFunction()
+  const gasPrice = await provider.getGasPrice()
+  const adjustedGasPrice = gasPrice.mul(GAS_MULTIPLIER)
+  return {
+    gasPrice: adjustedGasPrice,
+    gasLimit,
+  }
+}
+
+export async function airdrop(badgeCid: string, recipients: string[], shouldPumpGas = false) {
+  const provider = RpcService.getPolygonProvider()
+  const raftOwner = new ethers.Wallet(RAFT_OWNER_PK, provider)
+  const contract = new ethers.Contract(POLYGON_BADGES_CONTRACT_ADDRESS, BadgesAbi, raftOwner)
+  const ipfsAddress = `ipfs://${badgeCid}/metadata.json`
+  const formattedRecipients = checksumAddresses(recipients)
+  logger.log(`Airdropping, pumping gas ${shouldPumpGas}`)
+  let txn
+  if (shouldPumpGas) {
+    const gasConfig = await estimateGas(async () => contract.estimateGas.airdrop(formattedRecipients, ipfsAddress))
+    txn = await contract.connect(raftOwner).airdrop(formattedRecipients, ipfsAddress, gasConfig)
+  } else {
+    txn = await contract.connect(raftOwner).airdrop(formattedRecipients, ipfsAddress)
+  }
+  await txn.wait()
+  logger.log('Airdropped badge with txn hash:', txn.hash)
+  return txn.hash
+}
+
+export async function reinstateBadge(badgeId: string) {
+  const provider = RpcService.getPolygonProvider()
+  const raftOwner = new ethers.Wallet(RAFT_OWNER_PK, provider)
+  const contract = new ethers.Contract(POLYGON_BADGES_CONTRACT_ADDRESS, BadgesAbi, raftOwner)
```

Comment:
We could extract this first three lines to a `getBadgesContract` function or something like that right? Maybe we could have two. One to get the raftOwner and another one to get the contract as I see we are repeating these lines in all these functions.